pkg/integrations/v2/app_o11y_receiver/handler/handler.go

```diff
@@ -59,6 +59,12 @@ func (ar *AppO11yHandler) HTTPHandler(logger log.Logger) http.Handler {
 			return
 		}
 
+		// check API key if one is provided
+		if len(ar.config.APIKey) > 0 && r.Header.Get("x-api-key") != ar.config.APIKey {
```

Comment:
When comparing crypto strings, use [ContstantTimeCompare](https://pkg.go.dev/crypto/subtle#ConstantTimeCompare) to avoid [timing attacks](https://en.wikipedia.org/wiki/Timing_attack)

Reply:
good one, wasn't aware of this! done